File: test_lsp.py
#!/usr/bin/env python3
"""Test file for LSP configuration."""

import os
import sys
from typing import List, Optional


def greet(name: str) -> str:
    """Greet a person by name."""
    return f"Hello, {name}!"


def process_numbers(numbers: List[int]) -> Optional[float]:
    """Process a list of numbers and return their average."""
    if not numbers:
        return None
    
    # This line has a deliberate error for testing diagnostics
    total = sum(numbers)
    average = total / len(numbers)
    return average


def main() -> None:
    """Main function."""
    name = "World"
    greeting = greet(name)
    print(greeting)
    
    # Test with some numbers
    test_numbers = [1, 2, 3, 4, 5]
    avg = process_numbers(test_numbers)
    
    if avg is not None:
        print(f"Average: {avg:.2f}")
    
    # This will cause a type error for testing
    result = process_numbers("not a list")  # Type error here
    print(f"Result: {result}")


if __name__ == "__main__":
    main()

File: test_lsp.rs
// Test file for Rust LSP configuration

use std::collections::HashMap;

#[derive(Debug)]
struct Person {
    name: String,
    age: u32,
}

impl Person {
    fn new(name: &str, age: u32) -> Self {
        Person {
            name: name.to_string(),
            age,
        }
    }
    
    fn greet(&self) -> String {
        format!("Hello, my name is {} and I'm {} years old!", self.name, self.age)
    }
}

fn process_numbers(numbers: Vec<i32>) -> Option<f64> {
    if numbers.is_empty() {
        return None;
    }
    
    let sum: i32 = numbers.iter().sum();
    let avg = sum as f64 / numbers.len() as f64;
    Some(avg)
}

fn main() {
    let person = Person::new("Alice", 30);
    println!("{}", person.greet());
    
    let numbers = vec![1, 2, 3, 4, 5];
    match process_numbers(numbers) {
        Some(avg) => println!("Average: {:.2}", avg),
        None => println!("No numbers provided"),
    }
    
    // Test HashMap
    let mut scores = HashMap::new();
    scores.insert("Alice", 95);
    scores.insert("Bob", 87);
    
    for (name, score) in &scores {
        println!("{}: {}", name, score);
    }
    
    // This will cause a warning for testing diagnostics
    let unused_variable = 42;  // Warning: unused variable
}